src/tests/test_ucc_verifier.cpp

```diff
@@ -0,0 +1,153 @@
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/ucc/ucc_verifier/ucc_verifier.h"
+#include "config/indices/type.h"
+#include "datasets.h"
+
+namespace tests {
+namespace onam = config::names;
+
+class UCCVerifierSimpleParams {
+private:
+    algos::StdParamsMap params_map_;
+    size_t num_clusters_violating_ucc_ = 0;
+    size_t num_rows_violating_ucc_ = 0;
+    std::vector<model::PLI::Cluster> clusters_violating_ucc_;
+
+public:
+    UCCVerifierSimpleParams(config::IndicesType column_indices,
+                            size_t const num_clusters_violating_ucc,
+                            size_t const num_rows_violating_ucc,
+                            std::vector<model::PLI::Cluster> clusters_violating_ucc,
+                            std::string_view dataset, char const separator = ',',
+                            bool const has_header = true)
+        : params_map_({{onam::kCsvPath, test_data_dir / dataset},
+                       {onam::kSeparator, separator},
+                       {onam::kHasHeader, has_header},
+                       {onam::kEqualNulls, true}}),
+          num_clusters_violating_ucc_(num_clusters_violating_ucc),
+          num_rows_violating_ucc_(num_rows_violating_ucc),
+          clusters_violating_ucc_(std::move(clusters_violating_ucc)) {
+        // if column_indices is empty then it is not inserted into params_map_ (and this option is
+        // not passed to the algorithm), in this case default implementation will be used
+        if (!column_indices.empty()) {
+            params_map_.insert({onam::kUCCIndices, std::move(column_indices)});
+        }
+    }
+
+    algos::StdParamsMap GetParamsMap() const {
```

Comment:
```suggestion
    algos::StdParamsMap const& GetParamsMap() const {
```

Reply:
If we do this we will not able to pass this method return value into `CreateAndLoadAlgorithm(OptionMap&& options)` (because this function need r-value reference) directly, we need to create an additional variable. In the first review, it was proposed to [avoid this](https://github.com/Mstrutov/Desbordante/pull/290#discussion_r1404596051). As a result, add the `const` modifier?

Reply:
```cpp
template <typename T, typename OptionMap>
CreateAndLoadAlgorithm(OptionMap&& options)
```
`option` here is not passed by rvalue reference, but by ["universal" reference](https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers), so passing lvalues to it works perfectly fine.
Moreover, there is almost no difference from the caller point of view between `T const&` and `T` return types, in almost any context where `T` works `T const&` will also work

Reply:
Thanks for the article, the proposed version is implemented in the branch